cmd/vulcan-zap/main.go

```diff
@@ -114,51 +146,85 @@ func main() {
 			}
 		}
 
-		log.Println("Waiting for spider results...")
+		logger.Debug("Waiting for spider results...")
 		time.Sleep(5 * time.Second)
 
 		// Scan actively only if explicitly indicated.
 		if opt.Active {
-			log.Println("Running active scan...")
+			logger.Debug("Running active scan...")
 
 			resp, err = client.Ascan().Scan(targetURL.String(), "True", "False", "", "", "", "")
 			if err != nil {
-				return err
+				return fmt.Errorf("error executing the active scan: %v", err)
+			}
+
+			v, ok := resp["scan"]
+			if !ok {
+				return fmt.Errorf("scan is not present in response body when calling Ascan().Scan()")
+			}
+
+			scanid, ok := v.(string)
+			if !ok {
+				return errors.New("scan is present in response body when calling Ascan().Scan() but it is not a string")
 			}
 
-			scanid = resp["scan"].(string)
 			for {
 				time.Sleep(5 * time.Second)
+
 				ascan := client.Ascan()
-				resp, _ = ascan.Status(scanid)
 
-				progress, _ := strconv.Atoi(resp["status"].(string))
+				resp, err := ascan.Status(scanid)
+				if err != nil {
+					return fmt.Errorf("error getting the status of the scan: %v", err)
+				}
+
+				v, ok := resp["status"]
+				if !ok {
+					// In this case if we can not get the status let's fail.
+					return errors.New("can not retrieve the status of the scan")
+				}
+				status, ok := v.(string)
+				if !ok {
+					return errors.New("status is present in response body when calling Ascan().Scatus() but it is not a string")
+				}
+				progress, err := strconv.Atoi(status)
+				if err != nil {
+					return fmt.Errorf("can not convert status value %s into an int", status)
+				}
+
 				state.SetProgress((1 + float32(progress)) / 200)
-				log.Printf("Active scan at %v progress.", progress)
+
+				logger.Debugf("Active scan at %v progress.", progress)
 				if progress >= 100 {
 					break
 				}
 			}
 
-			log.Println("Waiting for active scan results...")
+			logger.Debug("Waiting for active scan results...")
 			time.Sleep(5 * time.Second)
 		}
 
 		// Retrieve alerts.
 		alerts, err := client.Core().Alerts("", "", "", "")
 		if err != nil {
-			return err
+			return fmt.Errorf("error retrieving alerts: %v", alerts)
 		}
 
-		alertsSlice := alerts["alerts"].([]interface{})
+		alertsSlice, ok := alerts["alerts"].([]interface{})
+		if !ok {
+			return errors.New("alerts does not exist or it is not an array of interface{}")
+		}
 
 		vulnerabilities := make(map[string]*report.Vulnerability)
 		for _, alert := range alertsSlice {
-			a := alert.(map[string]interface{})
+			a, ok := alert.(map[string]interface{})
+			if !ok {
+				return errors.New("alert it is not a map[string]interface{}")
+			}
 
 			v, err := processAlert(a)
 			if err != nil {
-				log.Println(err)
+				logger.WithError(err).Error("can not process alert")
```

Comment:
That is very much up for discussion since there is indeed an error in processing the specific alert, but I would just "Warn" here, since we are not returning any error and it's not something that taints the whole check.

Reply:
Done